cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------
+# Auto creation of CMake targets from pkg-config
+#
+# Copyright (C) 2017 Alexandre Pretyman. All rights reserved.
+#
+# Looks for a pkgconfig module of a given name and exports a cmake
+# target of the same name
+# ----------------------------------------------------------------------
+
+include(FindPkgConfig)
+
+include(hunter_internal_error)
+include(hunter_status_debug)
+
+function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
+  if(TARGET "${PKG_CONFIG_MODULE}")
+    return()
+  endif()
+  pkg_check_modules(${PKG_CONFIG_MODULE} ${PKG_CONFIG_MODULE})
```

Comment:
This function will find packages from Hunter because it respects CMAKE_PREFIX_PATH variable. But this functionality available since CMake 3.1:
* https://cmake.org/cmake/help/v3.1/module/FindPkgConfig.html
```
the CMAKE_PREFIX_PATH, CMAKE_FRAMEWORK_PATH, and
CMAKE_APPBUNDLE_PATH cache and environment variables will be
added to pkg-config search path
```
Not available in CMake 3.0:
* https://cmake.org/cmake/help/v3.0/module/FindPkgConfig.html

By the way further documentation reading, since CMake 3.6 we have imported targets from the box:
* https://cmake.org/cmake/help/v3.6/module/FindPkgConfig.html

```
The IMPORTED_TARGET argument will create an imported
target named PkgConfig::<PREFIX>> that can be passed directly
as an argument to target_link_libraries()
```

We have to use `cmake_minimum_required(VERSION)` or `if(CMAKE_VERSION VERSION_LESS)` here.